jupyter_ydoc/ydoc.py

```diff
@@ -10,87 +10,185 @@
+    def dirty(self) -> Optional[bool]:
+        """
+        Returns whether the document in memory has changes that are not on disk.
+
+        :return: Whether the document in memory has changes that are not on disk.
+        :rtype: bool | None
+        """
         return self._ystate["dirty"]
 
     @dirty.setter
     def dirty(self, value: bool) -> None:
+        """
+        Sets whether the document in memory has changes that are not on disk.
+
+        :param value: Whether the document in memory has changes that are not on disk.
+        :type value: bool
+        """
         with self._ydoc.begin_transaction() as t:
             self._ystate.set(t, "dirty", value)
 
     @property
-    def path(self) -> None:
+    def path(self) -> Optional[str]:
+        """
+        Returns document's path.
+
+        :return: Document's path.
+        :rtype: str | None
+        """
         return self._ystate.get("path")
 
     @path.setter
     def path(self, value: str) -> None:
+        """
+        Sets document's path.
+
+        :param value: Document's path.
+        :type value: str
+        """
         with self._ydoc.begin_transaction() as t:
             self._ystate.set(t, "path", value)
 
     @abstractmethod
     def get(self):
+        """
+        Returns the content of the document.
+
+        :return: Document's content.
+        :rtype: Any
+        """
         pass
 
     @abstractmethod
     def set(self, value):
+        """
+        Sets the content of the document.
+
+        :param value: The content of the document.
+        :type value: Any
+        """
         pass
 
     @abstractmethod
     def observe(self, callback):
+        """
+        Subscribe to document changes.
+
+        :param callback: Callback that will be called when the document changes.
+        :type callback: function
+        """
         pass
 
     def unobserve(self):
+        """
+        Unsubscribe to document changes.
+
+        This method removes all the callbacks.
+        """
         for k, v in self._subscriptions.items():
             k.unobserve(v)
         self._subscriptions = {}
 
 
 class YFile(YBaseDoc):
     """
-    YFile class
+    Extends :class:`YBaseDoc`, and represents a plain text document.
+
+    Schema:
+
+    .. code-block:: json
+
+        {
+            "state": YMap,
+            "source": YText
+        }
     """
 
     def __init__(self, *args, **kwargs):
         """
-        Constructor
+        Construct a YFile.
+
+        :param ydoc: The :class:`y_py.YDoc` that will hold the data of the document.
+        :type ydoc: :class:`y_py.YDoc`
         """
         super().__init__(*args, **kwargs)
         self._ysource = self._ydoc.get_text("source")
 
     def get(self):
         """
-        get source
+        Returns the content of the document.
+
+        :return: Document's content.
+        :rtype: str
         """
         return str(self._ysource)
 
     def set(self, value):
```

Comment:
Worth updating the type annotation (`str`)?

Reply:
What do you mean?